pallets/testnet/src/lib.rs

```diff
@@ -0,0 +1,153 @@
+        DidStatus(IdentityId, AccountId),
+        /// Shows the `DID` associated to the `AccountId`, and a flag indicates if that DID has a
+        /// valid CDD claim.
+        /// (Target DID, Target Account, a valid CDD claim exists)
+        CddStatus(Option<IdentityId>, AccountId, bool),
+    }
+);
+
+decl_storage! {
+    trait Store for Module<T: Trait> as testnet {
+    }
+}
+decl_error! {
+    pub enum Error for Module<T: Trait> {
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+
+        type Error = Error<T>;
+
+        fn deposit_event() = default;
+
+        /// Register a new did with a CDD claim for the caller.
+        #[weight = <T as Trait>::WeightInfo::register_did(secondary_keys.len() as u32)]
+        pub fn register_did(
+            origin,
+            uid: InvestorUid,
+            secondary_keys: Vec<SecondaryKey<T::AccountId>>,
+        ) {
+            let sender = ensure_signed(origin)?;
+            Identity::<T>::_register_did(sender.clone(), secondary_keys, Some(ProtocolOp::IdentityRegisterDid))?;
+
+            // Add CDD claim
+            let did = Identity::<T>::get_identity(&sender).ok_or_else(|| "DID Self-register failed")?;
+            let cdd_claim = Claim::CustomerDueDiligence(CddId::new(did, uid));
+            Identity::<T>::base_add_claim(did, cdd_claim, did, None);
+        }
+
+        /// Registers a new Identity for the `target_account` and issues a CDD claim to it.
+        /// The Investor UID is generated deterministically by the hash of the generated DID and
+        /// then we fix it to be compliant with UUID v4.
+        ///
+        /// # See
+        /// - [RFC 4122: UUID](https://tools.ietf.org/html/rfc4122)
+        ///
+        /// # Failure
+        /// - `origin` has to be a active CDD provider. Inactive CDD providers cannot add new
+        /// claims.
+        /// - `target_account` (primary key of the new Identity) can be linked to just one and only
+        /// one identity.
+        ///
+        /// # Weight
+        /// `7_000_000_000
+        #[weight = <T as Trait>::WeightInfo::mock_cdd_register_did()]
+        pub fn mock_cdd_register_did(origin, target_account: T::AccountId) {
+            let cdd_id = Identity::<T>::ensure_perms(origin)?;
+            let target_did = Identity::<T>::base_cdd_register_did(cdd_id, target_account, vec![])?;
+            let target_uid = confidential_identity::mocked::make_investor_uid(target_did.as_bytes());
+
+            // Add CDD claim for the target
+            let cdd_claim = Claim::CustomerDueDiligence(CddId::new(target_did, target_uid.clone().into()));
+            Identity::<T>::base_add_claim(target_did, cdd_claim, cdd_id, None);
+
+            Self::deposit_event(RawEvent::MockInvestorUIDCreated(target_did, target_uid.into()));
+        }
+
+        /// Emits an event with caller's identity.
+        /// NB: The weight is a placeholder
+        #[weight =
```

Comment:
Maybe add a placeholder weight function in `pallet-weights` and add it to `T::WeightInfo`?

Better to say "TODO" or "FIXME" instead of "NB", so that can be easily discovered later.